src/map.cpp

```diff
@@ -3100,37 +3071,44 @@ bool map::has_nearby_fire( const tripoint &p, int radius )
     return false;
 }
 
-void map::mop_spills( const tripoint &p ) {
+bool map::mop_spills( const tripoint &p )
+{
+    bool retval = false;
     auto items = i_at( p );
     for( auto it = items.begin(); it != items.end(); ) {
```

Comment:
That doesn't compile because of a missing bracket and `erase` only takes one argument, but in the spirit of your suggestion, I've modernized the loop similarly to how you depicted it.